### PR TITLE
handle stringification the same way for all vpc-cni configuration pro…

### PR DESCRIPTION
**Pull request by PeterKoegel**
A finding of #1044 was that the way the configuration properties for the vpc-cni plugin are converted to strings is not handled in the same way for all properties.

Configuration for the vpc-cni core addon is defined as following structure:

```
  const result: Values = {
    init: {
      env: {
        DISABLE_TCP_EARLY_DEMUX: props?.disableTcpEarlyDemux,
        ENABLE_V6_EGRESS: props?.enableV6Egress,
      }
    },
    env: {
      AWS_EC2_ENDPOINT: props?.awsEc2Endpoint,
      ADDITIONAL_ENI_TAGS: props?.additionalEniTags,
      .. some more properties
    },
    enableNetworkPolicy: props?.enableNetworkPolicy,
    enableWindowsIpam: props?.enableWindowsIpam
  };
```

Before that configuration is passed to the core addon undefined properties are removed and non-string properties are converted into strings by calling JSON.stringify().

For the "env" and "init.env" subobjects this is done in line 456 and 461 in a loop that iterates over all properties of this child objects, for the properties of the main object however JSON.stringify() is called when assigning the values in line 469 and 470.

This pull request changes that in a way that there is a recursive function "ConvertPropertiesToString" that handles the functionality described above by iterating over the whole configuration object including the embeded subobjects.

I verified that this code change does not introduce any functional changes by comparing the generated cloud formation output for different sets of parameters generated with and without this change and assure they are equal.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by PeterKoegel** [comment]
@shapirov103 : this PR is not fully tested by it would be interesting for me to know if this is the right approach to solve this issue.

File: lib/addons/vpc-cni/index.ts [hunk 1]
```diff
@@ -401,6 +401,23 @@ export class VpcCniAddOn extends CoreAddOn {
...
+function ConvertPropertiesToString(helmValues: Values): void {
+  Object.keys(helmValues).forEach(key => {
+    if (helmValues[key] === undefined) {
+      delete helmValues[key]
+    }
+    else if (typeof helmValues[key] === 'object'){
+      ConvertPropertiesToString(helmValues[key]);
+    }
+    else if (typeof helmValues[key] !== 'string'){
+      helmValues[key] = JSON.stringify(helmValues[key])
```
Comment (shapirov103) [inline lib/addons/vpc-cni/index.ts:416]: github action issue missing semicolon here and line 410

**Review by shapirov103 — COMMENTED** [review]
let me validate the e2e test and will merge after the minor code quality issues are addressed

**Comment by shapirov103** [comment]
e2e test failure unrelated to the changes. Will be addressed in a separate PR.

**Review by shapirov103 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**